src/generator/pollers.ts

```diff
@@ -53,50 +56,60 @@ type ${poller.name} struct {
 	pt pollingTracker
 }
 
-// Done returns true if the polling operation has terminated either in a success case or failure case,
-// otherwise it will return false
-func (p *${poller.name}) Done() bool {
-	return p.pt.hasTerminated()
-}
-
-// ID ... 
-func (p *${poller.name}) ID() string {
-	return ""
+// Poll returns false if there was an error or polling has reached a terminal state
+func (p *${poller.name}) Poll(ctx context.Context) bool {
+	if done, err := p.done(ctx); err != nil {
+		return false
+	} else {
+		return !done
+	}
 }
 
-func (p *${poller.name}) Poll(ctx context.Context) (*${responseType}, error) {
-	if done, err := p.done(ctx); !done || err != nil {
-		return nil, err
-	}
+// Response returns the latest response that is stored from the latest polling operation
```

Comment:
Does this actually work right now though?  If the poller is in a non-terminal state and I call `Response()` won't the response handler fail as there's no payload to unmarshal?  We don't have to fix this now but we do need a solution for it.

Reply:
According to me it does work that way. Because I have not changed the underlying functionality of the poller, in that it saves the current response to the response field on it. But I'll double check this by stepping through one of the tests in the testserver. p.response() returns the resp field that's on the poller.

Reply:
OH

Reply:
Ignore my previous comment, I understand what you mean now. Yes we need to think of a better solution for handling the response.

Reply:
One idea is to add a `LastResponse() *http.Response` method to the interface and rename `Response()` to `Result()`

Reply:
Yes that could work. Of course currently in the handleResponse method we always assign at a minimum the RawResponse to the RawResponse field in the envelope. Would that be enough? (I like your idea it's clean and clear, just wondering about what we already have)

Reply:
We could, it just feels a little clunky to me.  Plus I think we'd have to change the codegen for the responder to properly handle this case.  The three of us should probably have a (hopefully quick) chat about this scenario.

Reply:
Ok sounds good to me. I'll schedule a meeting for next week. For now I'll leave it as is.